### PR TITLE
feat: form validation with yup

### PR DESCRIPTION
**Pull request by KushChauhanSS**
- Added a Button component.
- Done form validation using yup.
- Added helper functions - hasErrors to check form fields for errors, , isTouched to check whether the field is touched or not and getError to generate error message.
- and created a design as per the gif provided in the task - https://redmine.successive.tech/issues/40766

### CODE REVIEW

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    disabled,
+    style,
+    value,
+    onClick,
+  } = props;
+
+  return (
+    <button
+      type={value === 'Submit' ? 'submit' : 'button'}
+      style={color === 'default' || (color === 'primary' && disabled === true) ? { ...styles.basicButton, ...style } : { ...styles.basicButton, ...styles.buttonEnabled, ...style }}
```
Comment (adilsuccessive) [inline src/components/Button/Button.jsx]: optimise this condition.

File: src/components/index.js [hunk 2]
```diff
@@ -2,3 +2,4 @@ export { TextField } from './TextField/index';
 export { Slider } from './Slider/index';
 export { SelectField } from './SelectField/index';
```
Comment (adilsuccessive) [inline src/components/index.js]: no need to add /index at the end

File: src/helpers/helper.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+export const hasErrors = (data) => {
+  const fields = Object.keys(data);
+  const errors = fields.map((field) => data[field].hasErrors);
+  if (errors.includes(true)) {
```
Comment (adilsuccessive) [inline src/helpers/helper.js]: optimise this no need of if

**Review by adilsuccessive — CHANGES_REQUESTED** [review]

**Comment by KushChauhanSS** [comment]
Done with the changes.

**Review by adilsuccessive — APPROVED** [review]

**Pull request merged**